modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationNotificationsUtil.java

```diff
@@ -662,69 +673,82 @@ public Void visitNamedListNode(String key, NamedListNode<?> newNamedList) {
      */
     private static Collection<DynamicConfiguration<InnerNode, ?>> mergeAnyConfigs(
             Collection<DynamicConfiguration<InnerNode, ?>> anyConfigs,
-            DynamicConfiguration<InnerNode, ?> anyConfig
+            @Nullable DynamicConfiguration<InnerNode, ?> anyConfig
     ) {
-        if (anyConfigs.isEmpty()) {
-            return List.of(anyConfig);
-        } else {
-            Collection<DynamicConfiguration<InnerNode, ?>> res = new ArrayList<>(anyConfigs.size() + 1);
-            
-            res.addAll(anyConfigs);
-            res.add(anyConfig);
-            
-            return res;
-        }
+        return Stream.concat(anyConfigs.stream(), Stream.of(anyConfig))
```

Comment:
I suggest not using streams here.

Reply:
why not?

Reply:
Code with streams:
```
return Stream.concat(anyConfigs.stream(), Stream.of(anyConfig))
        .filter(Objects::nonNull)
        .collect(Collectors.toList());
```
Code without streams:
```
var mergedConfigs = new ArrayList<DynamicConfiguration<InnerNode, ?>>(anyConfigs.size() + 1);

for (DynamicConfiguration<InnerNode, ?> config : anyConfigs) {
    if (config != null) {
        mergedConfigs.add(config);
    }
}

if (anyConfig != null) {
    mergedConfigs.add(anyConfig);
}

return mergedConfigs;
```
I think using streams is quite useful here

Reply:
I think it might be something like:
```
        if (anyConfig == null)
            return anyConfigs;
        else if (anyConfigs.isEmpty())
            return List.of(anyConfig);
        else {
            List<DynamicConfiguration<InnerNode, ?>> res = new ArrayList<>(anyConfigs.size() + 1);
            
            res.addAll(anyConfigs);
            res.add(anyConfig);
            
            return res;
        }
```

Reply:
Unfortunately it can't: `anyConfigs` can contain `null`